### DISCUSSION TITLE
How to specify "configure" options when installing a package (specifically xterm)

### DISCUSSION THREAD

**Discussion by gtackett** [Homebrew/discussions#1570] [Everyday usage]
I want to specify several options for the `configure` that is done when I install `xterm` (it's flush with configuration options) but I can't find way to do this successfully. The brew man page says this about `brew install`:

>    install [options] formula|cask [...]
>        Install a formula or cask. Additional options specific to a formula may be appended to the command.
> 

But if I try `$ brew install xterm --enable-trace` I just get:
`Error: invalid option: --enable-trace`

Perhaps this means the formula for `xterm` is missing something? I can take a shot at editing it, but although I've used `brew` many times to do simple (no-options) installation, I don't know know much about formulas, and the complete documentation for them is a lot to have to wade through. I don't need to know everything there is to know about formulas, after all.

After modifying the formula, I also don't know how to tell `brew` to install using my modified formula

**Comment by nandahkrishna** [comment]
> But if I try `$ brew install xterm --enable-trace` I just get:
> `Error: invalid option: --enable-trace`

Options are not supported when installing formulae from homebrew/core. If you want to specify options to `configure` when building the formula from source, you'll have to edit the formula as desired (add the options you want, such as `--enable-trace` to the `configure` step) and use `brew install --build-from-source <FORMULA-NAME>`. I wouldn't recommend modifying formulae in homebrew-core directly, if you intend to regularly use your modified formula.

For that, you may wish to create a personal [Tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap) for your modified formula. You may want to change the name of your modified formula, perhaps `xterm-modified`, if you do this.

**Reply by tlk** [reply]
Just to add to nandahkrishna's answer, I think the simplest way would be something like this:

```
mkdir -p experiment/Formula
cp $(brew --repo)/Library/Taps/homebrew/homebrew-core/Formula/xterm.rb experiment/Formula/xterm-modified.rb
cd experiment
brew install --formula Formula/xterm-modified.rb
```

**Reply by gtackett** [reply]
> Options are not supported when installing formulae from homebrew/core.

So is the man page for **brew** incorrect, or just confusing?

**Reply by nandahkrishna** [reply]
I should've been clearer: options to modify/enable/disable features in formulae are not supported for homebrew/core.

**Comment by gtackett** [comment]
> mkdir -p experiment/Formula
> cp $(brew --repo)/Library/Taps/homebrew/homebrew-core/Formula/xterm.rb experiment/Formula/xterm-modified.rb
> cd experiment
> brew install --formula Formula/xterm-modified.rb
> ```

Thanks, @tlk . That has worked well for me.
